File: parse-rss.js
const sqlite3 = require('sqlite3').verbose();
const fs = require('fs');

const db = new sqlite3.Database('/root/.config/rss-viewer/feeds.db');

// 今天 0:00:00 Asia/Shanghai -> UTC
const today = new Date('2026-08-21T00:00:00+08:00');
const now = new Date('2026-08-21T19:45:00+08:00');

const sentUrls = JSON.parse(fs.readFileSync('/root/.openclaw/rss-sent.json', 'utf8'));

const articles = [];

db.each(`
  SELECT a.title, a.link, a.published_at, f.url as feed_url, f.title as feed_title
  FROM articles a
  JOIN feeds f ON a.feed_id = f.id
  ORDER BY a.published_at DESC
`, (err, row) => {
  if (err) {
    console.error(err);
    return;
  }

  const pubDate = new Date(row.published_at);
  
  // 检查是否在今天范围内
  if (pubDate >= today && pubDate <= now) {
    // 检查是否已发送
    if (!sentUrls.includes(row.link)) {
      articles.push({
        title: row.title,
        link: row.link,
        pubDate: row.published_at,
        feedTitle: row.feed_title,
        feedUrl: row.feed_url
      });
    }
  }
}, (err) => {
  if (err) {
    console.error(err);
    return;
  }
  
  console.log(JSON.stringify(articles, null, 2));
  console.log(`\n共 ${articles.length} 篇新文章`);
  
  db.close();
});

File: parse_rss.py
import xml.etree.ElementTree as ET
from datetime import datetime, timezone, timedelta
import json

# Parse the RSS feed
tree = ET.parse('/root/.openclaw/workspace/feed.xml')
root = tree.getroot()

# Get today's date in Shanghai (UTC+8)
now = datetime.now(timezone(timedelta(hours=8)))
today_start = now.replace(hour=0, minute=0, second=0, microsecond=0)
today_start_utc = today_start.astimezone(timezone.utc)
now_utc = now.astimezone(timezone.utc)

# Read sent URLs
sent = set()
try:
    with open('/root/.openclaw/rss-sent.json', 'r') as f:
        sent = set(json.load(f))
except:
    pass

print(f"Now Shanghai: {now}")
print(f"Today start UTC: {today_start_utc}")
print(f"Now UTC: {now_utc}")
print(f"Sent count: {len(sent)}")

articles = []
for item in root.findall('.//item'):
    title = item.find('title').text or ''
    link = item.find('link').text or ''
    pub_date_el = item.find('pubDate')
    pub_date_str = pub_date_el.text if pub_date_el is not None else ''
    desc_el = item.find('description')
    desc = desc_el.text if desc_el is not None else ''
    
    # Parse pubDate
    try:
        dt = datetime.strptime(pub_date_str, '%a, %d %b %Y %H:%M:%S %z')
    except:
        try:
            dt = datetime.strptime(pub_date_str, '%a, %d %b %Y %H:%M:%S %Z')
            dt = dt.replace(tzinfo=timezone.utc)
        except:
            continue
    
    # Check if in today's range
    if today_start_utc <= dt <= now_utc:
        if link not in sent:
            # Extract source from description
            source = desc.split(']')[0].replace('[', '') if '[' in desc else link.split('/')[2]
            articles.append({
                'title': title,
                'link': link,
                'pubDate': pub_date_str,
                'source': source,
                'description': desc
            })

print(f"\nNew articles today: {len(articles)}")
for i, a in enumerate(articles):
    print(f"{i+1}. {a['title']}")
    print(f"   {a['link']}")
    print(f"   {a['pubDate']}")
    print()

# Save filtered articles for processing
import json
with open('/root/.openclaw/workspace/articles_today.json', 'w') as f:
    json.dump(articles, f, indent=2)
